Azure Nightly Builds fail to find the `versions.yaml` file

With the addition of the ability to read versions from the `versions.yaml` file currently the nightly builds fail as follows:

```bash
...
Error: open versions.yaml: no such file or directory
Error: Process completed with exit code 1.
```

source ^^ : https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6229388473/job/16907802165

My guess is that the current working directory is: `cloud-api-adaptor/azure/image`. So when accessing `versions.yaml` we should access it as `../../versions.yaml`.
https://github.com/confidential-containers/cloud-api-adaptor/blob/d64c9d4241a4ff34ccc4b855257c54c9b397bdd7/.github/workflows/azure-podvm-image-build.yml#L51-L63

Yes it fails because of the following error:

```bash
make image BINARIES=
  echo "successfully built $IMAGE_ID"
  echo "image-id=${IMAGE_ID}" >> "$GITHUB_OUTPUT"
  shell: /usr/bin/bash -e {0}
  env:
    AZURE_PODVM_IMAGE_DEF_NAME: podvm_image0
    AZURE_PODVM_IMAGE_VERSION: 202[3](https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6255429517/job/17008716701#step:16:3).09.21
    COMMUNITY_GALLERY_PREFIX: /CommunityGalleries/cocopodvm-d0e[4](https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6255429517/job/17008716701#step:16:4)f3[5](https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6255429517/job/17008716701#step:16:5)f-5530-4b9c-859[6](https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6255429517/job/17008716701#step:16:6)-11248[7](https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6255429517/job/17008716701#step:16:7)cdea[8](https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6255429517/job/17008716701#step:16:8)5
    PODVM_IMAGE_NAME: peerpod-image-625542[9](https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6255429517/job/17008716701#step:16:9)517-2
    SSH_USERNAME: peerpod
...
    PODVM_DISTRO: ubuntu
Error: unknown command ".cloudimg.ubuntu.focal.amd64.url | select(. != null)" for "yq"
Run 'yq --help' for usage.
../../podvm/../Makefile.defaults:[16](https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6255429517/job/17008716701#step:16:16): *** cloudimg.ubuntu.focal.amd64.url is not set, and could not be automatically set from /home/runner/work/cloud-api-adaptor/cloud-api-adaptor/cloud-api-adaptor/versions.yaml.  Stop.
Error: Process completed with exit code 2.
```

The generated path is weird: `/home/runner/work/cloud-api-adaptor/cloud-api-adaptor/cloud-api-adaptor/versions.yaml`.

--- a/.github/workflows/azure-podvm-image-build.yml
+++ b/.github/workflows/azure-podvm-image-build.yml
@@ -103,7 +103,9 @@ jobs:
       env:
         GOPATH: /home/runner/go
       if: steps.kata-agent-cache.outputs.cache-hit != 'true'
-      run: make "$(realpath -m ../../podvm/files/usr/local/bin/kata-agent)"
+      run: |
+        make "$(realpath -m ../../podvm/files/usr/local/bin/kata-agent)"
+        rm -f "${GOPATH}/bin/yq"
 
     - name: Set up pause cache
       id: pause-cache

--- a/Makefile.defaults
+++ b/Makefile.defaults
@@ -1,9 +1,15 @@
-ifeq (, $(shell command -v yq 2> /dev/null))
-$(error "No yq on PATH, consider doing snap install yq")
+YQ_COMMAND ?= yq
+VERSIONS_SRC := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))/versions.yaml
+
+ifeq (, $(shell command -v $(YQ_COMMAND) 2> /dev/null))
+$(error "$(YQ_COMMAND) not found, consider doing snap install yq")
 endif
 
-YQ_COMMAND     := yq
-VERSIONS_SRC := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))/versions.yaml
+YQ_MAJOR_VERSION := $(shell $(YQ_COMMAND) --version | sed -nE 's/.* v?([0-9]+).*/\1/p')
+
+ifneq (4, $(YQ_MAJOR_VERSION))
+$(error "yq major version should be 4, is: $(YQ_MAJOR_VERSION)")
+endif
 
 define query
 $(or $(shell $(YQ_COMMAND) '.$(1) | select(. != null)' $(VERSIONS_SRC)), \

